In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib import rc
rc('text', usetex=True)

def momento_inercia_experimental(T, K, Kerror):
    """La funcion devuelve el momento de inercia
    y el error a partir de:

    T: periodo
    K: constante recuperacion del muelle
    Kerror: error de la constante de recuperacion del muelle
    """    
    momento = T**2/(4*np.pi**2)*K
    error_momento = T**2/(4*np.pi**2)*Kerror
    
    return momento, error_momento


# Práctica 1: obtención de momentos de inercia

## 1. Determinación de la constante de recuperación del muelle espiral

Para la determinación de la constante de recuperación del muelle se ha seguido el método consistente
en relizar distintas mediciones de la fuerza $F$ necesaria para mantener la varilla en equilibro.
Todas las mediciones de la fuerza han sido realizadas colocando el dinamómetro a la misma distancia ($d=0,302$ m) 
del soporte para objetos del muelle espiral. En las distintas mediciones ha variado el ángulo $\phi$ de rotación del
muelle con respecto al ángulo de equilibro, se ha tomado medida de dicho ángulo visualmente. 

La siguiente tabla muestra los distintos valores obtenidos para la **fuerza** $F$ para cada uno de los
**ángulos** medidos para $\phi$. La tercera columna muestra el **momento de fuerza** $\Gamma=F·d$ sobre el punto donde de ha medido
la fuerza.
$$
\begin{array} {|r|r|r|}
\hline \phi\ (rad) & F\ (N) & \Gamma\ (N\cdot m) \\
\hline \pi & 0,40 & 0,1208 \\
\hline 2\pi & 0,60 & 0,1812 \\
\hline 3\pi & 1,10 & 0,3322 \\
\hline 4\pi & 1,25 & 0,3775 \\
\hline 5\pi & 1,55 & 0,4681 \\
\hline 6\pi & 1,70 & 0,5134 \\
\hline 
\end{array}
$$

In [4]:
# Obtención de la regresión
phi = np.pi*np.array([1,2,3,4,5,6])
d = 0.302
forces = np.array([0.4,0.6,1.1,1.25,1.55,1.7])
torque = d*forces
coeffs, residuals =  np.polyfit(phi, torque, deg=1, cov=True)

k, intercept = coeffs[0], coeffs[1]
k_error = np.sqrt(residuals)[0,0]
intercept_error = np.sqrt(residuals)[1,1]

print("Constante recuperación del muelle: ", k)
print("Ordenada en el origen: ", intercept)
print("Error constante del muelle: ", k_error)
print("Error ordenada en el origen: ", intercept_error)

# Plot de la regresión
plt.plot(phi, k*phi + intercept)
ax = plt.gca()
fig = plt.gcf()
plt.plot(phi, torque,'.')
ax.set_xlabel('$\phi\ (rad)$')
ax.set_ylabel('$\Gamma\ (N\cdot m)$')
fig.suptitle("$\Gamma\ vs.\ \phi$")
fig.savefig('regression_plot.png', dpi=150)
plt.close()

Constante recuperación del muelle:  0.026092316098894147
Ordenada en el origen:  0.045299999999999875
Error constante del muelle:  0.0021713460079364408
Error ordenada en el origen:  0.026565866033224632


Calculando la regresión  lineal:

$\Gamma = K \phi + b$ 

se obtienen los siguientes resultados:

Constante de recuperación:

$K \pm \Delta K = 0.026 \pm 0.002$

Ordenada en el origen:

$b \pm \Delta b = 0.045 \pm 0.026$

![Gráfico: ángulo vs. momento de fuerza](regression_plot.png)


## 2. Determinación experimental de momentos de inercia

La determinación experimental de los momentos de inercia se ha realizado utilizando ecuación
$$ I_z = K\frac{T^2}{4\pi^2} $$
que relaciona el momento de inercia  de los cuerpos $I_z$, la constante de recuperación del muelle $K$ obtenida en la sección anterior, y el periodo de oscilación $T$.
El periodo de oscilación $T$ hace referencia al periodo de oscilación del movimiento armónico simple que describe de manera aproximada la dinámica del sistema formado por el muelle espiral y el sólido acoplado a él.

Para cada uno de los sólidos se han realizado 3 mediciones de los semiperiodos. Las mediciones de los semiperiodos se han realizado rotando el muelle un cierto ángulo y liberando el sistema, dejando que este realizase 3 oscilaciones. Mediciones has sido anotadas en *segundos* con 3 cifras significtivas, habiendose redondeado la tercera cifra significativa.

Se ha estimado que el periodo asociado a cada cuerpo es la media de los 3 periodos obtenidos, denotada por $\overline{T}$.

El error en la determinación experimental del momento de inercia será igual a 
$\Delta I_z = \Delta K \frac{\overline{T}^2}{4\pi^2}$ .

### 2.1 Disco

Semiperiodos obtenidos: 

$$
\begin{array} {|r|r|r|r|}
\hline T/2\ (s) & 0,92 & 0,93 & 0,93 \\
\hline 
\end{array}
$$

Periodo medio: $\overline{T} = 1.85\ s$

#### Momento de inercia experimental:





In [13]:
T = 2*np.mean([.92,.93,.93])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: {:.3e}".format(T))
print("Momento de inercia: {:.3e}".format(momento))
print("Error momento de inercia: {:.3e}".format(error_momento))

Periodo: 1.853e+00
Momento de inercia: 2.270e-03
Error momento de inercia: 1.889e-04


$I_z \pm \Delta I_z = 2,270 \cdot 10^{-3} \pm 0,189\cdot 10^{-3}\ kg\cdot m^2 $

#### Momento de inercia teórico:

Las mediciones del sólido han dado los siguientes valores:

Masa: m = 0,358 kg

Diámetro: d = 0,224 m 

El momento de inercia teórico del disco es



In [11]:
m=0.358
d=0.224
momento_teorico_disco = 0.5 * m *(d/2)**2
print("Momento de inercia teórico del disco:  {:.3e}".format(momento_teorico_disco))

Momento de inercia teórico del disco:  2.245e-03


$I_z(disco) = \frac{1}{2}m (d/2)^2 = 2,245\cdot 10 ^{-3}\ kg\cdot m^2$



### 2.2 Varilla (sin masas)

Semiperiodos obtenidos: 

$$
\begin{array} {|r|r|r|}
\hline T/2\ (s) & 1,14 & 1,15 & 1,16 \\
\hline 
\end{array}
$$



In [10]:
T = 2*np.mean([1.14,1.15,1.16])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: ",T)
print("Momento de inercia: {0:.3e}".format(momento))
print("Error momento de inercia: {0:.3e}".format(error_momento))

Periodo:  2.3000000000000003
Momento de inercia: 3.496e-03
Error momento de inercia: 2.910e-04


Periodo medio: $\overline{T} = 2,3\ s$

#### Momento de inercia experimental:

$I_z \pm \Delta I_z = 3,496 \cdot 10^{-3} \pm 0.29 \cdot 10^{-3}\ kg\cdot m^2$

#### Momento de inercia teórico:

Las mediciones del sólido han dado los siguientes valores:

Masa:  m = 0,1312 kg

Longitud: l = 0,605 m



In [14]:
m=0.1312
l=0.605
momento_teorico_varilla = m * l**2/12
print("Momento de inercia teórico de la varilla: {:.3e}".format(momento_teorico_varilla) )


Momento de inercia teórico de la varilla: 4.002e-03


El momento de inercia teórico de la varilla es

$I_z = \frac{1}{12}m l^2 = 4,002 \cdot 10^{-3} \ kg\cdot m^2$

### 2.3 Cilindro macizo más base

Semiperiodos obtenidos: 

$$
\begin{array} {|r|r|r|}
\hline T/2\ (s) & 0,42 & 0,47 & 0,43 \\
\hline 
\end{array}
$$




In [15]:
T = 2*np.mean([0.42,0.47,0.43])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: ",T)
print("Momento de inercia: {:.3e}".format(momento))
print("Error momento de inercia: {:.3e}".format(error_momento))


Periodo:  0.8799999999999999
Momento de inercia: 5.118e-04
Error momento de inercia: 4.259e-05


Periodo medio: $\overline{T} = 0,88\ s$

#### Momento de inercia experimental:

$I_z \pm \Delta I_z = 5,112\cdot 10^{-4} \pm 0,460 \cdot 10^{-4}\ kg\cdot m^2 $

#### Momento de inercia teórico:



In [17]:
# Datos cilindro soporte
d_cil = 0.1
h_cil=0.005
# Datos  anillo soporte
d_in = 0.095
d_ext = 0.1
h_an=0.006
# Masa total soporte
m=0.109
# Volumenes cilindro y anillo
v_cil = np.pi*h_cil*(d_cil/2)**2
v_an = np.pi*h_an*((d_ext/2)**2-(d_in/2)**2)
v_total = v_cil+v_an
# Proporción cilindro y anillo
p_cil = v_cil/v_total
p_an = v_an/v_total
# Masa teórica cilindro y anillo
m_an = p_an*m
m_cil = p_cil*m
print("La masa del cilindro es {:.3e}:".format(m_cil))
print("La masa del anillo es: {:.3e}".format(m_an))
# Momentos teóricos cilindro y anillo
Iz_cil = 0.5*m_cil*(d_cil/2)**2
Iz_an = 0.5*m_an*(d_ext**2+d_in**2)
Iz_base = Iz_cil + Iz_an

print("El momento teórico del cilindro es: {:.3e}".format(Iz_cil))
print("El momento teórico del anillo es: {:.3e}".format(Iz_an))
print("El momento teórico del soporte es: {:.3e}".format(Iz_base))

La masa del cilindro es 9.758e-02:
La masa del anillo es: 1.142e-02
El momento teórico del cilindro es: 1.220e-04
El momento teórico del anillo es: 1.086e-04
El momento teórico del soporte es: 2.306e-04


#### Momento de inercia teórico:

Las mediciones del cilindro macizo han dado los siguientes valores:

Masa: m = 0,3461 kg

Diámetro: d = 0,089 m 

Se han tomando medidas para que permiten aproximar la base de sujección como un cilindro de altura $h_{cil}=0,005\ m$ y diámetro $d=0,100\ m$ más un anillo de altura $h_{an}=0,006\ m$ con diámetro exterior $d_{ext}=0,100\ m$ y diámetro interior $d_{int}=0,095$.
Suponiendo que la densidad de masa es uniforme, podemos calcular la masa del cilindro y el anillo que conforman la base como la masa total del conjunto multiplicada por la proporcion del volumen de cada elemento. La masa total de la base es $m=0,109\ kg$.

El volumen del cilindro es $V_{cilindro} = \pi\cdot h_{cil}(d/2)^2$.

El volumen del anillo es $V_{anillo} = \pi\cdot h_{an}((d_{ext}/2)^2-(d_{int}/2)^2)$.

El volumen total es $V_{total} = V_{anillo}+V_{cilindro}$, luego las masas del cilindro y del anillo 
son:

$m_{anillo} = m \cdot \frac{V_{anillo}}{V_{total}} = 0,0114\ kg$

$m_{cilindro} = m \cdot \frac{V_{cilindro}}{V_{total}} = 0,0976\ kg$.

Ahora podemos calcular el momento de inercia de la base de sujección como la suma de los momentos del anillo
y el cilindro que la componen:

$I_z(anillo) = \frac{1}{2} m_{anillo}\cdot((d_{in}/2)^2+(d_{ext}/2)^2) = 1,1 \cdot 10^{-4} kg\cdot m^2$.

$I_z(cilindro) = \frac{1}{2} m_{cilindro} (d/2)^2= 1,2 \cdot 10^{-4}$

$I_z(base) = I_z(anillo) + I_z(cilindro) = 2,3 \cdot 10^{-4}\ kg\cdot m^2.$



In [18]:
d_cilindro_macizo = 0.089
m_cilindro_macizo = 0.3461
momento_inercia_cilindro = 0.5 * m_cilindro_macizo * (d_cilindro_macizo/2)**2
print("El momento de inercio teórico del cilindro macizo es: {:.3e}".format(momento_inercia_cilindro))

El momento de inercio teórico del cilindro macizo es: 3.427e-04


El momento de inercia teórico del cilindro macizo es:

$I_z(cilindro\ macizo) =  \frac{1}{2}m\cdot(d/2)^2 = 3,4\cdot 10^{-4}\ kg\cdot m^2 $

y el momento de inercia del conjunto, mediante la aproximación realizada del momento de inercia de la base de sujección es

$I_z (conjunto) = I_z(cilindro\ macizo) + I_z(base)  = 5,7 \cdot 10^{-4}\ kg\cdot m^2$.

El momento de inercia experimental $5,1\cdot 10^{-4} \pm 0,4 \cdot 10^{-4}\ kg\cdot m^2$ debería ser similar al valor $I_z(conjunto)$. La diferencia de valores puede deberse a errores en mediciones o errores de redondeo.

### 2.4 Cilindro hueco

Semiperiodos obtenidos: 

$$
\begin{array} {|r|r|r|}
\hline T/2\ (s) & 0,44 & 0,46 & 0,48 \\
\hline 
\end{array}
$$






In [19]:
T = 2*np.mean([0.44,0.46,0.48])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: ",T)
print("Momento de inercia: {:.3e}".format(momento))
print("Error momento de inercia: {:.3e}".format(error_momento))

Periodo:  0.9199999999999999
Momento de inercia: 5.594e-04
Error momento de inercia: 4.655e-05


Periodo medio: $\overline{T} = 0,92\ s$

#### Momento de inercia experimental:

$I_z + \Delta I_z = 5,594\cdot 10^{-4} \pm 0,466\cdot 10^{-4} kg\cdot m^2 $



In [21]:
m_cilindro_hueco = 0.3413
d_exterior = 0.09
d_interior = 0.088
momento_cilindro_hueco = 0.5 * m_cilindro_hueco * ((d_interior/2)**2+(d_exterior/2)**2)
print("Momento teórico del cilindro hueco: {:.3e}".format(momento_cilindro_hueco))

Momento teórico del cilindro hueco: 6.759e-04


#### Momento de inercia teórico:

Masa: m = 0,3413 kg

Diámetro exterior: $d_e = 0,090\ m$

Diámetro interior: $d_i = 0,088\ m$

$I_z(cilindro\ hueco) = \frac{1}{2}m((d_e/2)^2+(d_i/2)^2)= 6,76 \cdot 10^{-4}\ kg\cdot m^2$

Si al momento de inercia del cilindro hueco, le añadimos el momento de inercia teórico de la base de sujección, calculado en la sección anterior, obtenemos el momento de inercia teórico del conjunto formado por la base más el cilindro hueco:

$I_z (conjunto) = I_z(cilindro\ hueco) + I_z(base)  = 9,06 \cdot 10^{-4}\ kg\cdot m^2$

El momento de inercia experimental $5,594 \cdot 10^{-4} \pm 0,466 \cdot 10^{-4}\ kg\cdot m^2$ debería ser similar al valor $I_z(conjunto)$. La diferencia de valores puede deberse a errores en las mediciones y redondeos en los cálculos.

### 2.5 Varilla con masas puntuales

Semiperiodos obtenidos: 

$$
\begin{array} {|r|r|r|}
\hline T/2\ (s) & 2,39 & 2,40 & 2,41 \\
\hline 
\end{array}
$$




In [22]:
T = 2*np.mean([2.39, 2.4, 2.41])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: ",T)
print("Momento de inercia: {:.3e}".format(momento))
print("Error momento de inercia: {:.3e}".format(error_momento))

Periodo:  4.8
Momento de inercia: 1.523e-02
Error momento de inercia: 1.267e-03


Periodo medio: $\overline{T} = 4,80\ s$

#### Momento de inercia experimental:

$I_z \pm \Delta I_z = 1.523\cdot 10^{-2} \pm 0,127\cdot 10^{-2} kg\cdot m^2 $



In [23]:
m_puntuales = 0.4741
distancia_centro = 0.17
momento_inercia_masas_puntuales = m_puntuales*distancia_centro**2
print("Momento de inercia teórico de las masa puntuales: {:.3e}".format(momento_inercia_masas_puntuales))

Momento de inercia teórico de las masa puntuales: 1.370e-02


#### Momento de inercia teórico:

Masas (se han pesado las dos masas juntas, asumimos que tienen exactamente la misma masa): m = 0,4742 kg.

Cada una de las masas es igual a $m_1=m_2=\frac{m}{2}$

Distancia de las masas al centro: 0,17 m

Calculamos el momento de inercia teórico de las dos masas puntuales como

$I_z(masas) = m_1 \cdot d^2 + m_2 \cdot d^2 = m \cdot d^2 = 1,370\cdot 10^{-2}\ kg\cdot m^2$

El momento de inercia teórico del conjunto formado por las dos masas puntuales junto a la varilla (el momento teórico de la varilla fue calculado anteriormente) es la suma

$I_z(masas\ y\ varilla) = I_z(masas) + I_z(varilla) = 1,370\cdot 10^{-2} + 0,400\cdot 10^{-2} = 1,770 \cdot 10^{-2}\ kg\cdot m^2$.

## 3. Determinación del momento de inercia con respecto a un eje que no pasa por el centro de gravedad



Para la determinación experimental del momento de inercia con respecto a un eje que no pasa por el centro de gravedad se han llevado a cabo dos experimentos, en el primero se ha sujetado la varilla al centro de rotación del muelle por uno de los extremos de la varilla; en el segundo experimento, se ha sujetado la varilla por un punto que está a una distancia del extremo de la varilla igual a $\frac{1}{4}L$, siendo $L=0,605\ m$ la longitud de la varilla. Se recuerda que la masa de la varilla es $m=0,1312\ kg$.

Se han anotado los semiperiodos con 4 cifras significativas. 

### Primer experimento 

El centro de gravedad está a una distancia $L/2$ del centro de rotación del muelle, y se obtienen los siguientes semiperiodos:
$$
\begin{array} {|r|r|r|}
\hline T/2\ (s) & 2,338 & 2,332 & 2,326 \\
\hline 
\end{array}
$$



In [24]:
T = 2*np.mean([2.338, 2.332, 2.326])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: ",T)
print("Momento de inercia: {:.3e}".format(momento))
print("Error momento de inercia: {:.3e}".format(error_momento))

Periodo:  4.664000000000001
Momento de inercia: 1.438e-02
Error momento de inercia: 1.196e-03


Periodo medio: $\overline{T} = 4,664\ s$

#### Momento de inercia experimental:

$I_z + \Delta I_z = 1,438\cdot 10^{-2} \pm 0,120\cdot 10^{-2} kg\cdot m^2 $



In [25]:
masa_varilla = 0.1312
longitud_varilla = 0.605
momento_teorico_exp_1 = masa_varilla * longitud_varilla**2 / 3
print("Momento de inercia teórico (experimento 1): {:.3e}".format(momento_teorico_exp_1))

Momento de inercia teórico (experimento 1): 1.601e-02


#### Momento de inercia teórico

El momento de inercia teórico en este caso es 

$I_z = \int_0^{L} x^2 \frac{m}{L}\,dx = \frac{m\cdot L^2}{3} = \frac{0,1312\cdot 0,605^2}{3} =  1,6\cdot 10^{-2}kg\cdot m^2$

### Segundo experimento

El centro de gravedad está a una distancia $L/4$ del centro de rotación del muelle, y se obtienen lo siguientes semiperiodos:
$$
\begin{array} {|r|r|r|}
\hline T/2\ (s) & 1,858 & 1,832 & 1,811 \\
\hline 
\end{array}
$$



In [26]:
T = 2*np.mean([1.858, 1.832, 1.811])
momento, error_momento = momento_inercia_experimental(T, k, k_error)
print("Periodo: ",T)
print("Momento de inercia: {:.3e}".format(momento))
print("Error momento de inercia: {:.3e}".format(error_momento))

Periodo:  3.6673333333333336
Momento de inercia: 8.889e-03
Error momento de inercia: 7.397e-04


Periodo medio: $\overline{T} = 3,667\ s$

#### Momento de inercia experimental:

$I_z + \Delta I_z = 8,889\cdot 10^{-3} \pm 0,740\cdot 10^{-3} kg\cdot m^2 $



In [27]:
momento_teorico_exp_2 = masa_varilla * longitud_varilla**2 / 9
print("Momento de inercia teórico (experimento 2): {:.3e}".format(momento_teorico_exp_2))

Momento de inercia teórico (experimento 2): 5.336e-03


El momento de inercia teórico en este caso es 

$I_z =\int_0^{L/4} x^2 \frac{m}{L}\,dx + \int_0^{3L/4} x^2 \frac{m}{L}\,dx =\frac{m\cdot L^2}{9} = \frac{0,1312\cdot 0,605^2}{9}=  5,34\cdot 10^{-3}kg\cdot m^2$

## 3. Variación del momento de inercia en función de la distancia de las masas

En este experimento se calcula el momento de inercia de la varilla con dos masas puntuales colocadas a diferentes distancias respecto al centro de rotación del muelle. 

En la siguiente tabla se recogen los datos del experimento: la primera columna contiene la distancia $d$ al centro de la varilla de cada una de las masas puntuales (las dos masas están a la misma distancia del centro de la varilla), la segunda, tercera y cuarta columna recogen las mediciones de los semiperiodos, la quinta columna recoge la media de los semiperiodos multiplicada por 2, valor que es empleado como medición del periodo, la última columna contiene el valor experimental obtenido para el momento de inercia (no se incluyen los errores).

$$
\begin{array} {|r|r|r|}
\hline d\ (m) & T_1/2\ (s) & T_2/2\ (s) & T_3/2\ (s)& \overline{T}\ (s) & I_z\ (kg\cdot m^2)\\
\hline 0,066 & 1,404       & 1,415      & 1,424     & 2,828          & 5,28\cdot 10^{3}   \\
\hline 0,116 & 1,813       & 1,834      & 1,852     & 3,666          & 8,88\cdot 10^{3}   \\
\hline 0,166 & 2,650       & 2,630      & 2,620     & 5,266          & 18,33\cdot 10^{3}   \\
\hline 0,216 & 3,240       & 3,230      & 3,220     & 6,460          & 27,58\cdot 10^{3}   \\
\hline 0,266 & 4,012       & 3,976      & 3,946     & 7.956          & 41,84\cdot 10^{3}   \\
\hline 
\end{array}
$$






In [28]:
# Obtención de la regresión
T = [2.828, 3.666, 5.266, 6.460, 7.956]
inertia = [momento_inercia_experimental(t, k, k_error)[0] for t in T]
L = 0.032
d = 0.05
ds = np.array([1,2,3,4,5])*d+L/2

coeffs_inertia, residuals_inertia =  np.polyfit(ds**2, inertia, deg=1, cov=True)

masas, inercia_varilla = coeffs_inertia[0], coeffs_inertia[1]
masas_error = np.sqrt(residuals_inertia)[0,0]
inercia_varilla_error = np.sqrt(residuals_inertia)[1,1]


# Plot de la regresión
plt.plot(ds**2, inertia,'.')
ax = plt.gca()
fig = plt.gcf()

ax.set_xlabel('$d^2\ (m^2)$')
ax.set_ylabel('$I_z\ (kg\cdot m^2)$')
fig.suptitle("$d^2\ vs.\ I_z$")
ax.plot(ds**2, masas*ds**2+inercia_varilla)
fig.savefig('inertia_vs_distance.png', dpi=150)
plt.close()

Ajustando los datos a la recta de ecuación

$I_z = M\cdot d + b$

se obtiene

$M \pm \Delta M = 0,555 \pm 0,015\ kg$,

$b \pm \Delta b = 2,33 \cdot 10^{-3} \pm 0,63\cdot 10^{-3}\ kg\cdot m^2$.

Por tanto, la estimación experimental de la suma de las masas $m_1+m_2$ obtenida es $0,555 \pm 0,015\ kg$, y la estimación experimental del momento de inercia de la varilla es $2,33 \cdot 10^{-3} \pm 0,63\cdot 10^{-3}\ kg\cdot m^2$.

El siguiente gráfico muestra la distancia de cada una de las masas al centro frente al momento de inercia.

![caption](inertia_vs_distance.png)